modules/mesos/src/main/java/org/apache/ignite/mesos/resource/IgniteProvider.java

```diff
@@ -17,28 +17,51 @@
 
 package org.apache.ignite.mesos.resource;
 
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import java.io.BufferedReader;
 import java.io.File;
 import java.io.FileOutputStream;
 import java.io.IOException;
+import java.io.InputStreamReader;
 import java.net.HttpURLConnection;
 import java.net.URL;
 import java.nio.channels.Channels;
 import java.nio.file.Files;
 import java.nio.file.Path;
 import java.nio.file.Paths;
 import java.nio.file.StandardCopyOption;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import java.util.stream.Collectors;
 import org.apache.ignite.mesos.ClusterProperties;
 
+import static org.apache.ignite.mesos.ClusterProperties.IGNITE_VERSION;
+
 /**
  * Class downloads and stores Ignite.
  */
 public class IgniteProvider {
+    /** Logger. */
+    private static final Logger log = Logger.getLogger(IgniteProvider.class.getSimpleName());
+
     // This constants are set by maven-ant-plugin.
     /** */
-    private static final String DOWNLOAD_LINK = "http://ignite.run/download_ignite.php";
+    private static final String DOWNLOAD_URL_PATTERN = "";
```

Comment:
Still, maybe I will keep this one as https://archive.apache.org/dist/ignite/%s/apache-ignite-fabric-%s-bin.zip, access it in case that it's not possible to detect latest version - if something changes on Apache side. We should be able to access current version as a version of ignite-mesos JAR. I'm just worried that we're changing one fragile thing (ignite.run) to another fragile thing (apache mirror JSON API).

Reply:
@alamar Totally share your concerns on the fragility.
But anyway, user will see errors if Apache changes something, and can specify the version (from the "latest" to "2.7.0" for example) to download it from the archives. The url is in pom.

Reply:
I'll put it back, in case it's needed for development.